support/build/Makefile.rules

```diff
@@ -252,21 +252,30 @@ verbose_cmd = @printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 verbose_cmd_inner = printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 endif
 
+# Writes arguments into a build_args file
+#
+# write_build_args $target,$longargs(optional)
+ifeq ($(CONFIG_BUILD_ARGS_FILE),y)
+write_build_args = $(if $(2),$(file > $(addsuffix .build_args,$(1)),$(2))@$(addsuffix .build_args,$(1)),)
+else
+write_build_args = $(2)
```

Comment:
Why do we use $(2) here?

Reply:
If `BUILD_ARGS_FILE` is disabled, the `$longargs` will not be written into the file but directly appended to the argument. So `write_build_args` is either the name of the file were `$longargs` is written to (`BUILD_ARGS_FILE`) or directly the `$longargs` (`!BUILD_ARGS_FILE`).

Reply:
Makes sense. Thanks!